src/adagrams.js

```diff
@@ -1,8 +1,134 @@
-const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
-  },
-};
+class Adagrams  {
+  static drawLetters() {
+      const AllLetters = ['a','a','a','a','a','a','a','a','a','b','b','c','c','d','d','d','d','e','e','e','e','e','e','e','e','e','e','e','e','f','f','g','g','g','h','h','i','i','i','i','i','i','i','i','i','j','k','l','l','l','l','m','m','n','n','n','n','n','n','o','o','o','o','o','o','o','o','p','p','q','r','r','r','r','r','r','s','s','s','s','t','t','t','t','t','t','u','u','u','u','v','v','w','w','x','y','y','z'];
+      const handy = [];
+      for (let i = 1; i < 11; i++) {
+        let index = [Math.floor(Math.random()*AllLetters.length)];
+        let removed = AllLetters.splice(index, 1);
+        let removed2 = removed.pop()
+        handy.push(removed2.toUpperCase());
+      }
+      return handy;
+  }
+  static usesAvailableLetters (input, lettersInHand) {
+    let value = true;
+    [...input].forEach(letter => {
+      // ugh forEach CAN"T BE ESCAPED WITH A RETURN OR BREAK
+      // for(x in y) use in for objects
+      // for(x of y) for strings or arrays or anything else
+      let index = lettersInHand.indexOf(letter);
+      if (index === -1) {
+        value = false;
+      } else {
+        lettersInHand.splice(index, 1);
+      }
+    })
+    return value;
+  }
+  static scoreWord (word) {
+    const tile = {
+      'a': 1,
+      'b': 3,
+      'c': 3,
+      'd': 2,
+      'e': 1,
+      'f': 4,
+      'g': 2,
+      'h': 4,
+      'i': 1,
+      'j': 8,
+      'k': 5,
+      'l': 1,
+      'm': 3,
+      'n': 1,
+      'o': 1,
+      'p': 3,
+      'q': 10,
+      'r': 1,
+      's': 1,
+      't': 1,
+      'u': 1,
+      'v': 4,
+      'w': 4,
+      'x': 8,
+      'y': 4,
+      'z': 10,
+    }
+
+    // so I know I'm updating score, but I"M JUST GOING TO UPDATE IT so shouldn't I be allowed to use a const?
+    let score = 0;
+
+    [...word].forEach(letter => {
+      letter = letter.toLowerCase();
+      // WHY CAN"T I USE tile.letter?
+      // Cause its a string 'b' instead of b right?
+      // is there a good work around?
```

Comment:
Yes! it's because `letter` is a variable. dot notation doesn't work with variables, it is looking for a member named whatever that thing is (aka `letter`). `[]` notation is necessary for this kind of thing